src/app/GitCommands/Git/GitModule.cs

```diff
@@ -3191,18 +3185,45 @@ public IReadOnlyList<string> GetFullTree(string id)
                 .Split(Delimiters.NullAndLineFeed);
         }
 
-        public IEnumerable<INamedGitItem> GetTree(ObjectId? commitId, bool full)
+        public IEnumerable<IObjectGitItem> GetTree(ObjectId? commitId, bool full, CancellationToken cancellationToken = default)
+            => GetGitItemTree(commitId, full, "", cancellationToken);
+
+        public IEnumerable<IObjectGitItem> GetGitItemTree(ObjectId? commitId, bool full, string? fileName, CancellationToken cancellationToken = default)
         {
-            GitArgumentBuilder args = new("ls-tree")
+            bool isArtificial = commitId?.IsArtificial is true;
+            if (isArtificial && !full)
             {
-                "-z",
-                { full, "-r" },
-                commitId
-            };
+                throw new ArgumentOutOfRangeException(nameof(full), "Artificial commit requires 'full'.");
+            }
 
-            string tree = _gitExecutable.GetOutput(args, cache: GitCommandCache);
+            if (commitId == ObjectId.CombinedDiffId)
+            {
+                throw new ArgumentOutOfRangeException(nameof(commitId), "Cannot use CombinedDiffId.");
+            }
 
-            return _gitTreeParser.Parse(tree);
+            GitArgumentBuilder args = isArtificial
+                ? new("ls-files")
+                {
+                    // ls-files with same format as ls-tree
+                    "-z",
+                    { commitId == ObjectId.IndexId, "--cached", "--no-cached" },
+                    @$"--format=""{_gitTreeParser.GitTreeFormat}""",
+                    { !string.IsNullOrWhiteSpace(fileName), "--" },
```

Comment:
```suggestion
                    "--",
```
is slightly clearer for git (less ambiguous)

Reply:
Both ways are used in GiModule, I will change the others too
The code is cleaner, but reading the log I find this slightly confusing: should the be an argument following?

Reply:
I remember a bug which was resolved by appending the "--".